app/controllers/recipes_foods_controller.rb

```diff
@@ -0,0 +1,48 @@
+class RecipesFoodsController < ApplicationController
+  def index
+    @recipe_food = RecipeFood.new
+    @recipe_food.build_food
+  end
+
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
+    food_id = recipe_food_params[:food_id]
+    existing_food = @recipe.recipe_foods.find_by(food_id:)
+    if existing_food
+      existing_food.quantity += recipe_food_params[:quantity].to_i
+      if existing_food.save
+        redirect_to recipe_path(@recipe), notice: 'Food quantity updated successfully.'
+      else
+        redirect_to recipe_path(@recipe), alert: 'Failed to update food quantity.'
+      end
+    else
+      @recipe_food = RecipeFood.new(recipe_food_params)
+      @recipe_food.recipe = @recipe
+
+      if @recipe_food.save
+        redirect_to recipe_path(@recipe), notice: 'RecipeFood was successfully created'
+      else
+        render :new, alert: 'Something went wrong'
+      end
+    end
+  end
```

Comment:
1. In the create action, consider breaking down the logic into smaller methods or extracting some of the logic into the model to improve readability and maintainability.
2. In the create action, you are using redirect_to with a notice or alert message. It's good practice to use flash messages instead of including messages in the redirect. You can set flash messages in the controller and then display them in the view.


**Below is my suggestion;**
```
class RecipesFoodsController < ApplicationController
  # existing actions...

  def create
    @recipe = Recipe.find(params[:recipe_id])
    food_id = recipe_food_params[:food_id]
    existing_food = @recipe.recipe_foods.find_by(food_id: food_id)

    if existing_food
      update_existing_food(existing_food)
    else
      create_new_recipe_food
    end
  end

  def destroy
    @recipe_food = RecipeFood.find(params[:id])
    @recipe = @recipe_food.recipe

    update_recipe_totals_on_delete

    @recipe_food.destroy
    redirect_to recipe_path(@recipe), notice: 'RecipeFood deleted successfully!'
  end

  private

  def update_existing_food(existing_food)
    existing_food.quantity += recipe_food_params[:quantity].to_i

    if existing_food.save
      redirect_to recipe_path(@recipe), notice: 'Food quantity updated successfully.'
    else
      redirect_to recipe_path(@recipe), alert: 'Failed to update food quantity.'
    end
  end

  def create_new_recipe_food
    @recipe_food = RecipeFood.new(recipe_food_params)
    @recipe_food.recipe = @recipe

    if @recipe_food.save
      update_recipe_totals_on_create
      redirect_to recipe_path(@recipe), notice: 'RecipeFood was successfully created'
    else
      render :new, alert: 'Something went wrong'
    end
  end

  def update_recipe_totals_on_create
    @recipe.increment!(:foodcounter)
  end

  def update_recipe_totals_on_delete
    amount = @recipe.totalvaluecounter - @recipe_food.totalvalue
    @recipe.update(totalvaluecounter: amount)
    @recipe.decrement!(:foodcounter)
  end
end
```

Reply:
Thank you